File: cpp/leetcode/387.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        unordered_map <char, int> words;
        for(char ch : s) words[ch] ++;
        for(int i = 0; i < s.size(); i++){
            if(words[s[i]] == 1) return i;
        }
        return -1;
    }
};

File: python/leetcode/387.py
class Solution(object):
    def firstUniqChar(self, s):
        dic = {}
        for item in s:
            dic[item] = dic.get(item, 0) + 1
        for i, item in enumerate(s):
            if dic[item] < 2: return i
        return -1

class Solution(object):
    def firstUniqChar(self, s):
        for x in s:
            if s.find(x) == s.rfind(x):
                return s.find(x)
        return -1